specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-05-31/consumption.json

```diff
@@ -0,0 +1,334 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2018-05-31",
+    "title": "ConsumptionManagementClient",
+    "description": "Consumption management client provides access to consumption resources for Azure Enterprise Subscriptions."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Consumption/forecasts": {
+      "get": {
+        "tags": [
+          "Forecasts"
+        ],
+        "x-ms-odata": "#/definitions/Forecast",
+        "operationId": "Forecasts_List",
+        "description": "Lists the forecast charges by subscriptionId.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/consumption/"
+        },
+        "x-ms-examples": {
+          "ForecastsList": {
+            "$ref": "./examples/ForecastsList.json"
+          },
+          "ForecastsListFilterByGrain": {
+            "$ref": "./examples/ForecastsListFilterByGrain.json"
+          },
+          "ForecastsListFilterByDates": {
+            "$ref": "./examples/ForecastsListFilterByDates.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "$filter",
+            "description": "May be used to filter forecasts by properties/usageDate (Utc time), properties/chargeType or properties/grain. The filter supports 'eq', 'lt', 'gt', 'le', 'ge', and 'and'. It does not currently support 'ne', 'or', or 'not'.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/ForecastsListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
```

Comment:
even though this is not a pagable operation, adding following annotation

```
        "x-ms-pageable": {
          "nextLinkName": null
        }
```

will ensure the generator flatten the `value` field, thus generated method will have List as return type instead of a model type that composes list.

Reply:
Sure, applied the change